core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -961,6 +961,14 @@ public XContentBuilder dateValueField(XContentBuilderString rawFieldName, XConte
         return this;
     }
 
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, TimeValue timeValue) throws IOException {
```

Comment:
there's no need to duplicate the functionality here, this can call:

```java
timeValueField(rawFieldName, readableFieldName, timeValue.millis(), TimeUnit.MILLIS);
```